app/uk/gov/hmrc/vatsignupfrontend/controllers/principal/partnerships/ConfirmLimitedPartnershipController.scala

```diff
@@ -73,34 +77,43 @@ class ConfirmLimitedPartnershipController @Inject()(val controllerComponents: Co
     val optPartnershipUtr = request.session.get(SessionKeys.partnershipSautrKey).filter(_.nonEmpty)
     val optCompanyName = request.session.get(SessionKeys.companyNameKey).filter(_.nonEmpty)
     val optCompanyNumber = request.session.get(SessionKeys.companyNumberKey).filter(_.nonEmpty)
+    val optPartnershipType = request.session.get(SessionKeys.partnershipTypeKey).filter(_.nonEmpty)
 
     authorised() {
-      (optVatNumber, optPartnershipUtr, optCompanyName, optCompanyNumber) match {
-        case (Some(vatNumber), Some(partnershipUtr), Some(_), Some(_)) =>
+      (optVatNumber, optPartnershipUtr, optCompanyName, optCompanyNumber, optPartnershipType) match {
+        case (Some(vatNumber), Some(partnershipUtr), Some(_), Some(_), Some(partnershipType)) =>
         confirmPartnershipForm.bindFromRequest().fold(
-            error => Future.successful(BadRequest(confirm_partnership_utr(partnershipUtr, limitedPartnershipName = optCompanyName, error, routes.ConfirmLimitedPartnershipController.submit())))
+            error => Future.successful(BadRequest(
+              confirm_partnership_utr(partnershipUtr, limitedPartnershipName = optCompanyName, error, routes.ConfirmLimitedPartnershipController.submit())
+            ))
             , {
-              case Yes =>
-                storePartnershipInformationService.storePartnershipInformation(vatNumber, partnershipUtr, companyNumber = optCompanyNumber) flatMap {
+              case Yes => {
+                storePartnershipInformationService.storePartnershipInformation(
+                  vatNumber = vatNumber,
+                  sautr = partnershipUtr,
+                  companyNumber = optCompanyNumber,
+                  partnershipEntity = Some(BusinessEntitySessionFormatter.toBEPartnershipString(partnershipType))
```

Comment:
As it currently is, the acceptance tests fail. After making the suggested change, the technical difficulties page no longer display after confirming the partnership utr.

```suggestion
                  partnershipEntity = Some(partnershipType)
```